### ISSUE TITLE
can not reproduce the results

### ISSUE THREAD

**Issue by junleiz** [fxmeng/filter-grafting#5]
![image](https://user-images.githubusercontent.com/26025961/77132458-8a898e80-6a9a-11ea-9782-4398f851b752.png
I trained the resnet32 on cifar100 with 2 grafting setting. I set the seed 1 for the first model and 2 for the second. But I found the acc even decreased.

etwork:2    epoch:0    accuracy:14.700    best:14.700
Network:2    epoch:1    accuracy:16.540    best:16.540
Network:2    epoch:2    accuracy:24.700    best:24.700
Network:2    epoch:3    accuracy:23.520    best:24.700
Network:2    epoch:4    accuracy:32.250    best:32.250
Network:2    epoch:5    accuracy:34.070    best:34.070
Network:2    epoch:6    accuracy:37.280    best:37.280
Network:2    epoch:7    accuracy:37.320    best:37.320
Network:2    epoch:8    accuracy:36.540    best:37.320
Network:2    epoch:9    accuracy:40.050    best:40.050
Network:2    epoch:10    accuracy:44.270    best:44.270
Network:2    epoch:11    accuracy:39.200    best:44.270
Network:2    epoch:12    accuracy:38.870    best:44.270
Network:2    epoch:13    accuracy:44.300    best:44.300
Network:2    epoch:14    accuracy:43.690    best:44.300
Network:2    epoch:15    accuracy:38.960    best:44.300
Network:2    epoch:16    accuracy:47.530    best:47.530
Network:2    epoch:17    accuracy:40.490    best:47.530
Network:2    epoch:18    accuracy:46.240    best:47.530
Network:2    epoch:19    accuracy:40.100    best:47.530
Network:2    epoch:20    accuracy:41.650    best:47.530
Network:2    epoch:21    accuracy:44.300    best:47.530
Network:2    epoch:22    accuracy:44.440    best:47.530
Network:2    epoch:23    accuracy:44.870    best:47.530
Network:2    epoch:24    accuracy:43.860    best:47.530
Network:2    epoch:25    accuracy:46.760    best:47.530
Network:2    epoch:26    accuracy:32.770    best:47.530
Network:2    epoch:27    accuracy:42.220    best:47.530
Network:2    epoch:28    accuracy:48.920    best:48.920
Network:2    epoch:29    accuracy:44.960    best:48.920
Network:2    epoch:30    accuracy:45.670    best:48.920
Network:2    epoch:31    accuracy:45.630    best:48.920
Network:2    epoch:32    accuracy:45.840    best:48.920
Network:2    epoch:33    accuracy:46.910    best:48.920
Network:2    epoch:34    accuracy:51.240    best:51.240
Network:2    epoch:35    accuracy:48.490    best:51.240
Network:2    epoch:36    accuracy:49.460    best:51.240
Network:2    epoch:37    accuracy:45.080    best:51.240
Network:2    epoch:38    accuracy:49.390    best:51.240
Network:2    epoch:39    accuracy:45.370    best:51.240
Network:2    epoch:40    accuracy:40.510    best:51.240
Network:2    epoch:41    accuracy:39.560    best:51.240
Network:2    epoch:42    accuracy:46.540    best:51.240
Network:2    epoch:43    accuracy:48.780    best:51.240
Network:2    epoch:44    accuracy:49.220    best:51.240
Network:2    epoch:45    accuracy:46.590    best:51.240
Network:2    epoch:46    accuracy:40.120    best:51.240
Network:2    epoch:47    accuracy:44.470    best:51.240
Network:2    epoch:48    accuracy:42.030    best:51.240
Network:2    epoch:49    accuracy:47.310    best:51.240
Network:2    epoch:50    accuracy:46.580    best:51.240
Network:2    epoch:51    accuracy:45.010    best:51.240
Network:2    epoch:52    accuracy:46.270    best:51.240
Network:2    epoch:53    accuracy:47.070    best:51.240
Network:2    epoch:54    accuracy:46.270    best:51.240
Network:2    epoch:55    accuracy:49.480    best:51.240
Network:2    epoch:56    accuracy:45.360    best:51.240
Network:2    epoch:57    accuracy:46.950    best:51.240
Network:2    epoch:58    accuracy:47.840    best:51.240
Network:2    epoch:59    accuracy:52.580    best:52.580
Network:2    epoch:60    accuracy:43.420    best:52.580
Network:2    epoch:61    accuracy:68.060    best:68.060
Network:2    epoch:62    accuracy:68.230    best:68.230
Network:2    epoch:63    accuracy:68.550    best:68.550
Network:2    epoch:64    accuracy:68.750    best:68.750
Network:2    epoch:65    accuracy:68.490    best:68.750
Network:2    epoch:66    accuracy:68.290    best:68.750
Network:2    epoch:67    accuracy:68.320    best:68.750
Network:2    epoch:68    accuracy:67.870    best:68.750
Network:2    epoch:69    accuracy:68.240    best:68.750
Network:2    epoch:70    accuracy:67.790    best:68.750
Network:2    epoch:71    accuracy:67.170    best:68.750
Network:2    epoch:72    accuracy:68.130    best:68.750
Network:2    epoch:73    accuracy:68.610    best:68.750
Network:2    epoch:74    accuracy:66.910    best:68.750
Network:2    epoch:75    accuracy:66.640    best:68.750
Network:2    epoch:76    accuracy:66.710    best:68.750
Network:2    epoch:77    accuracy:66.220    best:68.750
Network:2    epoch:78    accuracy:65.440    best:68.750
Network:2    epoch:79    accuracy:66.520    best:68.750
Network:2    epoch:80    accuracy:66.810    best:68.750
Network:2    epoch:81    accuracy:66.030    best:68.750
Network:2    epoch:82    accuracy:65.430    best:68.750
Network:2    epoch:83    accuracy:66.470    best:68.750
Network:2    epoch:84    accuracy:66.250    best:68.750
Network:2    epoch:85    accuracy:65.690    best:68.750
Network:2    epoch:86    accuracy:65.500    best:68.750
Network:2    epoch:87    accuracy:66.020    best:68.750
Network:2    epoch:88    accuracy:65.160    best:68.750
Network:2    epoch:89    accuracy:63.700    best:68.750
Network:2    epoch:90    accuracy:65.590    best:68.750
Network:2    epoch:91    accuracy:65.310    best:68.750
Network:2    epoch:92    accuracy:63.440    best:68.750
Network:2    epoch:93    accuracy:64.340    best:68.750
Network:2    epoch:94    accuracy:64.090    best:68.750
Network:2    epoch:95    accuracy:64.020    best:68.750
Network:2    epoch:96    accuracy:63.130    best:68.750
Network:2    epoch:97    accuracy:62.210    best:68.750
Network:2    epoch:98    accuracy:63.610    best:68.750
Network:2    epoch:99    accuracy:63.960    best:68.750
Network:2    epoch:100    accuracy:64.730    best:68.750
Network:2    epoch:101    accuracy:65.030    best:68.750
Network:2    epoch:102    accuracy:64.990    best:68.750
Network:2    epoch:103    accuracy:64.130    best:68.750
Network:2    epoch:104    accuracy:63.280    best:68.750
Network:2    epoch:105    accuracy:63.800    best:68.750
Network:2    epoch:106    accuracy:64.050    best:68.750
Network:2    epoch:107    accuracy:63.460    best:68.750
Network:2    epoch:108    accuracy:64.790    best:68.750
Network:2    epoch:109    accuracy:64.470    best:68.750
Network:2    epoch:110    accuracy:65.210    best:68.750
Network:2    epoch:111    accuracy:64.350    best:68.750
Network:2    epoch:112    accuracy:62.980    best:68.750
Network:2    epoch:113    accuracy:63.390    best:68.750
Network:2    epoch:114    accuracy:63.860    best:68.750
Network:2    epoch:115    accuracy:64.430    best:68.750
Network:2    epoch:116    accuracy:62.950    best:68.750
Network:2    epoch:117    accuracy:63.950    best:68.750
Network:2    epoch:118    accuracy:64.000    best:68.750
Network:2    epoch:119    accuracy:63.570    best:68.750
Network:2    epoch:120    accuracy:62.570    best:68.750
Network:2    epoch:121    accuracy:70.290    best:70.290
Network:2    epoch:122    accuracy:69.990    best:70.290
Network:2    epoch:123    accuracy:70.000    best:70.290
Network:2    epoch:124    accuracy:70.210    best:70.290
Network:2    epoch:125    accuracy:69.750    best:70.290
Network:2    epoch:126    accuracy:69.850    best:70.290
Network:2    epoch:127    accuracy:70.200    best:70.290
Network:2    epoch:128    accuracy:69.730    best:70.290
Network:2    epoch:129    accuracy:69.830    best:70.290
Network:2    epoch:130    accuracy:69.780    best:70.290
Network:2    epoch:131    accuracy:69.520    best:70.290
Network:2    epoch:132    accuracy:69.560    best:70.290
Network:2    epoch:133    accuracy:69.630    best:70.290
Network:2    epoch:134    accuracy:69.770    best:70.290
Network:2    epoch:135    accuracy:69.750    best:70.290
Network:2    epoch:136    accuracy:69.390    best:70.290
Network:2    epoch:137    accuracy:69.630    best:70.290
Network:2    epoch:138    accuracy:69.250    best:70.290
Network:2    epoch:139    accuracy:69.460    best:70.290
Network:2    epoch:140    accuracy:69.420    best:70.290
Network:2    epoch:141    accuracy:69.230    best:70.290
Network:2    epoch:142    accuracy:69.490    best:70.290
Network:2    epoch:143    accuracy:69.430    best:70.290
Network:2    epoch:144    accuracy:69.220    best:70.290
Network:2    epoch:145    accuracy:69.660    best:70.290
Network:2    epoch:146    accuracy:69.330    best:70.290
Network:2    epoch:147    accuracy:69.070    best:70.290
Network:2    epoch:148    accuracy:69.260    best:70.290
Network:2    epoch:149    accuracy:69.350    best:70.290
Network:2    epoch:150    accuracy:69.130    best:70.290
Network:2    epoch:151    accuracy:69.270    best:70.290
Network:2    epoch:152    accuracy:68.890    best:70.290
Network:2    epoch:153    accuracy:69.220    best:70.290
Network:2    epoch:154    accuracy:68.980    best:70.290
Network:2    epoch:155    accuracy:68.850    best:70.290
Network:2    epoch:156    accuracy:68.970    best:70.290
Network:2    epoch:157    accuracy:69.260    best:70.290
Network:2    epoch:158    accuracy:69.140    best:70.290
Network:2    epoch:159    accuracy:69.100    best:70.290
Network:2    epoch:160    accuracy:68.860    best:70.290
Network:2    epoch:161    accuracy:68.990    best:70.290
Network:2    epoch:162    accuracy:69.120    best:70.290
Network:2    epoch:163    accuracy:68.780    best:70.290
Network:2    epoch:164    accuracy:69.190    best:70.290
Network:2    epoch:165    accuracy:68.560    best:70.290
Network:2    epoch:166    accuracy:68.860    best:70.290
Network:2    epoch:167    accuracy:68.860    best:70.290
Network:2    epoch:168    accuracy:68.620    best:70.290
Network:2    epoch:169    accuracy:69.010    best:70.290
Network:2    epoch:170    accuracy:68.760    best:70.290
Network:2    epoch:171    accuracy:68.680    best:70.290
Network:2    epoch:172    accuracy:68.950    best:70.290
Network:2    epoch:173    accuracy:68.830    best:70.290
Network:2    epoch:174    accuracy:68.740    best:70.290
Network:2    epoch:175    accuracy:68.780    best:70.290
Network:2    epoch:176    accuracy:68.620    best:70.290
Network:2    epoch:177    accuracy:68.410    best:70.290
Network:2    epoch:178    accuracy:68.540    best:70.290
Network:2    epoch:179    accuracy:68.660    best:70.290
Network:2    epoch:180    accuracy:68.610    best:70.290
Network:2    epoch:181    accuracy:68.750    best:70.290
Network:2    epoch:182    accuracy:68.690    best:70.290
Network:2    epoch:183    accuracy:68.730    best:70.290
Network:2    epoch:184    accuracy:68.710    best:70.290
Network:2    epoch:185    accuracy:68.750    best:70.290
Network:2    epoch:186    accuracy:68.590    best:70.290
Network:2    epoch:187    accuracy:68.710    best:70.290
Network:2    epoch:188    accuracy:68.740    best:70.290
Network:2    epoch:189    accuracy:68.690    best:70.290
Network:2    epoch:190    accuracy:68.820    best:70.290
Network:2    epoch:191    accuracy:68.820    best:70.290
Network:2    epoch:192    accuracy:68.480    best:70.290
Network:2    epoch:193    accuracy:68.810    best:70.290
Network:2    epoch:194    accuracy:68.760    best:70.290
Network:2    epoch:195    accuracy:68.790    best:70.290
Network:2    epoch:196    accuracy:68.770    best:70.290
Network:2    epoch:197    accuracy:68.590    best:70.290
Network:2    epoch:198    accuracy:68.730    best:70.290
Network:2    epoch:199    accuracy:68.840    best:70.290

The final act is 68.84. The corresponding baseline in your paper is 69.82

**Comment by junleiz** [comment]
The acc for the network1 is almost the same

**Comment by fxmeng** [comment]
![image](https://user-images.githubusercontent.com/60565778/77140577-02fd4900-6ab5-11ea-8052-81fae5c14dd7.png)
[1.log](https://github.com/fxmeng/filter-grafting/files/4358144/1.log)
Thanks for your attention,
First, because the learning rate drops every 60 epochs, the final result is not necessarily the best, and the best results often appear after each decline in the learning rate. As shown in the figure, the best result is 71.05 at epoch: 126. In your results, the best result of grafting should be 70.29, which is higher than the baseline.
However, your results are still lower than ours, maybe because you modified the code, please use our original version to experiment; or the environment is different, please compare the results of baseline on your environment.

**Comment by junleiz** [comment]
Thank you for your reply. I changed two things:

1. I set the batch_size for train_loader and valid_loader the same;
2. I set the random seed. Otherwise, it is hard to compare with baseline since their seeds may be totally different

I would try the origin version.
